### PR TITLE
[13.0][FIX] account: don't update date on nondraft move lines

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
If not, some things could change fiscal period, and we need to avoid that.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/migrations/13.0.1.1/post-migration.py [hunk 1]
```diff
@@ -288,7 +288,7 @@ def migration_invoice_moves(env):
                 THEN ail.currency_id ELSE NULL END, ail.partner_id, ail.uom_id,
                 ail.product_id, ail.account_analytic_id, ail.display_type,
                 ail.is_rounding_line, COALESCE(ai.move_id, am.id), ail.id, ai.date_invoice,
-                ail.create_uid, ail.create_date, ail.write_uid, ail.write_date
+                COALESCE(ai.date, ail.create_uid), ail.create_date, ail.write_uid, ail.write_date
```
Comment (pedrobaeza) [inline addons/account/migrations/13.0.1.1/post-migration.py]: The coalesce should be in the previous field, not here

File: addons/account/migrations/13.0.1.1/post-migration.py [hunk 2]
```diff
@@ -324,7 +324,7 @@ def migration_invoice_moves(env):
         THEN ail.currency_id ELSE NULL END, ail.partner_id, ail.uom_id,
         ail.product_id, ail.account_analytic_id, ail.display_type,
         ail.is_rounding_line, COALESCE(ai.move_id, am.id), ail.id, ai.date_invoice,
-        ail.create_uid, ail.create_date, ail.write_uid, ail.write_date
+        COALESCE(ai.date, ail.create_uid), ail.create_date, ail.write_uid, ail.write_date
```
Comment (pedrobaeza) [inline addons/account/migrations/13.0.1.1/post-migration.py]: The same

File: addons/account/migrations/13.0.1.1/post-migration.py [hunk 3]
```diff
@@ -360,7 +360,7 @@ def migration_invoice_moves(env):
         ait.amount, ait.currency_id, ait.base, ait.tax_id,
         ait.account_analytic_id, COALESCE(ai.move_id, am.id),
         ait.id, TRUE, am.state, 1.0, ai.commercial_partner_id, ai.date_invoice,
-        ait.create_uid, ait.create_date, ait.write_uid, ait.write_date
+        COALESCE(ai.date, ait.create_uid), ait.create_date, ait.write_uid, ait.write_date
```
Comment (pedrobaeza) [inline addons/account/migrations/13.0.1.1/post-migration.py]: The same

File: addons/account/migrations/13.0.1.1/post-migration.py [hunk 4]
```diff
@@ -497,11 +497,11 @@ def migration_voucher_moves(env):
         INSERT INTO account_move_line (company_id, account_id,
         sequence, name, quantity, price_unit, price_subtotal,
         product_id, analytic_account_id, move_id, old_voucher_line_id,
-        create_uid, create_date, write_uid, write_date)
+        date, create_uid, create_date, write_uid, write_date)
         SELECT avl.company_id, avl.account_id, avl.sequence, avl.name,
         avl.quantity, avl.price_unit, avl.price_subtotal, avl.product_id,
         avl.account_analytic_id, COALESCE(av.move_id, am.id), avl.id,
-        avl.create_uid, avl.create_date, avl.write_uid, avl.write_date
+        COALESCE(am.date, avl.create_uid), avl.create_uid, avl.create_date, avl.write_uid, avl.write_date
```
Comment (pedrobaeza) [inline addons/account/migrations/13.0.1.1/post-migration.py]: The same

File: addons/account/migrations/13.0.1.1/post-migration.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply addons/account/migrations/13.0.1.1/post-migration.py]: Yes, sorry, my mistake, for going too fast.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**